src/main/java/task/Parser.java

```diff
@@ -0,0 +1,93 @@
+                    if (completedIndex < 0 || completedIndex >= taskList.getCounter()) {
+                        throw new DukeException("OOPS!!! Index not found.");
+                    }
+
+                    taskList = taskList.doneTask(completedIndex);
+                } else if (textInput.startsWith("todo")) {
+                    if (textInput.equals("todo") || textInput.equals("todo ")) {
+                        throw new DukeException("OOPS!!! The description of a todo cannot be empty.");
+                    }
+                    String description = textInput.replaceFirst("todo ", "");
+                    taskList = taskList.addTask(new Todo(description));
+                } else if (textInput.startsWith("deadline")) {
+                    if (textInput.equals("deadline") || textInput.equals("deadline ")) {
+                        throw new DukeException("OOPS!!! The description of a deadline cannot be empty.");
+                    }
+                    String removeTaskWord = textInput.replaceFirst("deadline ", "");
+                    String[] taskSplit = removeTaskWord.split(" /by ");
+                    taskList = taskList.addTask(new Deadline(taskSplit[0], taskSplit[1]));
+                } else if (textInput.startsWith("event")) {
+                    if (textInput.equals("event") || textInput.equals("event ")) {
+                        throw new DukeException("OOPS!!! The description of a event cannot be empty.");
+                    }
+
+                    String removeTaskWord = textInput.replaceFirst("event ", "");
+                    String[] taskSplit = removeTaskWord.split(" /at ");
+                    taskList = taskList.addTask(new Event(taskSplit[0], taskSplit[1]));
+                } else if (textInput.startsWith("delete")) {
+                    if (textInput.equals("delete") || textInput.equals("delete ")) {
+                        throw new DukeException("OOPS!!! Index required.");
+                    }
+
+                    int deletedIndex = Integer.parseInt(textInput.replaceFirst("delete ", "")) - 1;
+                    if (deletedIndex < 0 || deletedIndex >= taskList.getCounter()) {
+                        throw new DukeException("OOPS!!! Index not found.");
+                    }
+
+                    taskList = taskList.deleteTask(deletedIndex);
+                } else if (textInput.startsWith("find")) {
+                    if (textInput.equals("find") || textInput.equals("find ")) {
+                        throw new DukeException("OOPS!!! Keyword required!");
+                    }
+
+                    String keyWord = textInput.replaceFirst("find ", "");
+                    taskList.findTask(keyWord);
+                } else {
+                    throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(");
+                }
+            } catch (DukeException e) {
+                Ui.printException(e);
+            } finally {
```

Comment:
I think this code structure is clean (referring to the one big try catch block) . I should adopt for my own use! good job Joel.